feat: bestiary tracker

omarcopires opened the pull request:
<!-- Note: Lines with this <!-- syntax are comments and will not be visible in
     your pull request. You can safely ignore or remove them. -->

### Pull Request Prelude

<!-- Thank you for working on improving The Forgotten Server! -->
<!-- Please complete these steps and check the following boxes by putting an `x`
     inside the [brackets] before filing your Pull Request. -->

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed <!-- Describe the changes that this pull request makes. -->
<img width="174" height="78" alt="image" src="https://github.com/user-attachments/assets/49d02929-bc78-413b-8dd3-4795193af6a8" />

<!-- You can safely ignore the links below:  -->
[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

omarcopires committed:
feat: bestiary tracker

ranisalt commented on data/lib/core/player.lua:
```diff
@@ -769,3 +769,85 @@ do
+
+function Player.getTrackedBestiary(self, isLogin)
+	if isLogin == nil then
+		isLogin = false
```
`isLogin` is unused

ranisalt commented on data/lib/core/player.lua:
Instead of adding tracked race id to storage k, use `base + raceId` and store 0 or 1, I think it will greatly simplify

omarcopires committed:
chore: remove unused variable

omarcopires committed:
refactor: optimize bestiary tracking storage

- Replace sequential storage with direct race ID mapping
- Improve performance with O(1) access to tracked creatures
- Add caching to prevent repeated storage scans
- Maintain backward compatibility with existing API

omarcopires committed:
feat: add table utility functions

- Create table.lua with common table operations
- Add table.contains for value existence check
- Add table.indexOf for finding element position
- Add table.slice for array slicing
- Include table.lua in core initialization

omarcopires committed:
fix: replace SQL placeholders with string.format in bestiary tracker queries

ranisalt commented on data/lib/core/player.lua:
```diff
@@ -769,3 +769,79 @@ do
+		table.insert(trackedBestiary, raceId)
+	elseif not checked and index then
+		table.remove(trackedBestiary, index)
+	end
```
I was thinking in having a big sparse table and operating like this:


```suggestion
    trackedBestiary[raceId] = checked or nil
```

(`or nil` to remove it instead of just setting to false)

Can we rely directly on storage values instead, and drop this table?

ranisalt commented on data/lib/core/player.lua:
```diff
@@ -769,3 +769,79 @@ do
+end
+
+function Player.getTrackedBestiaryCount(self)
+	local count = 0
```
`count` is unused


```suggestion
```

ranisalt commented on data/creaturescripts/scripts/login.lua:
```diff
@@ -33,6 +33,12 @@ function onLogin(player)
 	end
 
+	-- Bestiary tracker
+	local trackedBestiary = player:getTrackedBestiary(true)
```
```suggestion
	local trackedBestiary = player:getTrackedBestiary()
```

I think the parameter was removed at some point

ranisalt commented on data/lib/core/player.lua:
```diff
@@ -769,3 +769,79 @@ do
+		msg:addU32(kills)
+
+		local monsterType = MonsterType(raceId)
+		local info = monsterType and monsterType:getBestiaryInfo() or {prowess = 1, expertise = 2, mastery = 3}
```
Add a warning if the monster type is not existent, it should only happen if a creature is removed but data was not cleared
